### PR TITLE
Comment on deploy checklist when a pull request is CP'd

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
1. Comments on the deploy checklist like so when a pull request is CP'd:

	  ---
	  
	  :clap: Heads up @Expensify/applauseleads :clap:
	  
	  A [new pull request](https://github.com/Expensify/App/pull/7802) has been :cherries: cherry-picked :cherries: to staging, and will be deployed to staging in version `1.1.39-2` :rocket:
	  
	  ---

1. Then waits for the staging deploy for that version to complete

1. then comments again like so:

	  ---
	  
	  :rocket: _All set?…You bet!_ https://github.com/Expensify/App/releases/tag/1.1.39-2 has been deployed to staging :rocket:
	  
	  ---

### Fixed Issues
$ https://github.com/Expensify/App/issues/7178

### Tests
Unit tests added.

### QA Steps
1. Merge a PR with the `CP Staging` label with the checklist locked.
1. Verify that the first comment appears on the checklist and is correctly formatted.
1. Verify that the staging deploy completes, then the second comment appears on the checklist and is correctly formatted.
1. Merge another PR without the `CP Staging` label with the checklist locked.
1. Verify that no comments appear on the checklist.
1. Manually CP the PR.
1. Repeat steps 2 and 3.

### CODE REVIEW

**Comment by roryabraham** [comment]
This failure looks like it might be a bug in action-lint. Going to check it out more tomorrow.

**Comment by roryabraham** [comment]
cc @mountiny in case you want to review as well since you've taken an interesting in our GitHub Actions. Don't want to hold on that because I know you're only working ~30% right now.

**Comment by roryabraham** [comment]
Only added `CP Staging` so we can start QA upon merge.

**Review by mountiny — COMMENTED** [review]
Thank you for the ping indeed, I am very happy to learn from how others will review it too. It is still interesting to me how much we can do through the actions.

I did not see any problems with the code and the unit tests probably show it works well. Can't say much about best practices yet.

File: .github/workflows/preDeploy.yml [hunk 1]
```diff
@@ -157,6 +160,33 @@ jobs:
...
+        with:
+          GITHUB_TOKEN: ${{ secrets.OS_BOTIFY_TOKEN }}
+          TAG: ${{ env.NEW_VERSION }}
+
+      - name: Comment in StagingDeployCash to alert Applause that cherry-picked pull request has been deployed.
+        if: ${{ env.DO_CHERRY_PICK }}
+        run: |
+          gh issue comment \
+          "$(gh issue list --label StagingDeployCash --json number --jq '.[0].number')" \
+          --body ":tada: All set?…You bet! https://github.com/Expensify/App/releases/tag/${{ env.NEW_VERSION }} has been deployed to staging :tada:"
```
Comment (AndrewGable) [inline .github/workflows/preDeploy.yml:186]: Should this actually tag Applause leads? I don't think it does at the moment?

Comment (roryabraham) [reply .github/workflows/preDeploy.yml]: Fixed!

**Review by AndrewGable — APPROVED** [review]

**Comment by AndrewGable** [comment]
Feel free to merge when e2e is done 👍

**Comment by botify** [comment]
@roryabraham looks like this was merged without passing tests. Please add a note explaining why this was done and remove the `Emergency` label if this is not an emergency.

**Comment by roryabraham** [comment]
Not sure why it said it was merged without passing tests ... pretty sure I waited for tests to pass first 🤔

**Comment by luacmartins** [comment]
Got assigned QA, @roryabraham is there a proactive way of testing this or should I just monitor PRs that are merged with the `CP staging` label/manually CP'd?

**Comment by timszot** [comment]
I would say this works just fine per [this comment](https://github.com/Expensify/App/issues/7759#issuecomment-1048345774) that happened on the current checklist.

**Comment by luacmartins** [comment]
Checking this off since it was already deployed in Feb!

**Pull request merged**